### DISCUSSION TITLE
Is there documentation for Avalonia.Markup.Declarative?

### DISCUSSION THREAD

**Discussion by stevensrmiller** [AvaloniaUI/Avalonia#16790] [Q&A]
I am loving Avalonia and look forward to using it a lot. My work calls for everything to be done in C#, without XAML. That's going very nicely, but it seems to me that the methods in Avalonia.Markup.Declarative would be of great help. Is there documentation for it?

**Comment by Yricky** [comment]
Currently, there isn't one, perhaps compose is a better choice.

**Reply by stevensrmiller** [reply]
I love Avalonia. I have been able to do everything I need, and more, without XAML. Here's a bunch of examples I've published: https://github.com/stevensrmiller/AvaloniaWithoutXAML It's been pretty popular. I don't need to learn another library.

**Reply by duongphuhiep** [reply]
Love your repo, very useful to me. But be careful about imperative UI, it is not scale very well. If the UI is simple, then it is ok, but  when the UI become complex then all the imperative C# codes will be hard to maintain (and to theme). I don't like XAML, unfortunately, Avalonia doesn't support declarative UI with C#

**Comment by stevensrmiller** [comment]
Glad you like it!

Stevens
